### PR TITLE
Fix builds for Rails integration tests

### PR DESCRIPTION
**Pull request by mec**
A couple of things changed recently that broke these for us:

- Rails 7.2 beta dropped support for Ruby < 3.1
- ffi dropped support for Rubygems < 3.3.22

To fix this we have switched from fetching the Rails version to
supplying them in the GitHub actions, at least until Rails 7.2 is out.

We have also updated Rubygems and bundler to new enough versions to
compile the ffi gem.

We also had to add the build-essential package into a later build step
as ffi seems to get built a second time - this is possibly incorrect and
will need looking into.

We should possibly decide to at least drop testing for end of life versions:

- Ruby 2.7.8
- Rails 5.2.8.1

### CODE REVIEW

File: .github/workflows/rails-integration-tests.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
         name: Compute outputs
         run: |
           rails_versions=$(curl https://rubygems.org/api/v1/versions/rails.json | jq 'group_by(.number[:1])[] | (.[0].number) | select(.[:1]|tonumber > 4)' | jq -s -c)
-          echo "RAILS_VERSIONS=$rails_versions" >> $GITHUB_OUTPUT
+          echo 'RAILS_VERSIONS=["5.2.8.1","6.1.7.8","7.1.3.4"]' >> $GITHUB_OUTPUT
```
Comment (pezholio) [inline .github/workflows/rails-integration-tests.yml:21]: We probably want to do this on the line above to get the latest non-beta versions:

```
rails_versions=$(curl https://rubygems.org/api/v1/versions/rails.json | jq '[.[] | select(.number | test("beta") | not)] | group_by(.number[:1])[] | (.[0].number) | select(.[:1]|tonumber > 4)' | jq -s -c)
```

This outputs:

```
["5.2.8.1","6.1.7.8","7.1.3.4"]
```

**Review by pezholio — COMMENTED** [review]
Just one comment, I'd feel happier if we carried on getting the latest Rails versions when running the tests. I've made a comment that tweaks the JQ command to filter out the Beta versions (which I assume was why you ended up hardcoding 😃 )

File: .github/workflows/rails-integration-tests.yml (hunk 1 above)
Comment (mec) [reply .github/workflows/rails-integration-tests.yml]: Sure, I do prefer the more obvious way in this context (code that is not touched often and so the speed someone can undestand what is going on becomes more important), but let's get this out - updated. 👍

**Comment by mec** [comment]
Classic! So in the time it's taken me to get to this, Rails 7.2 is out so we need to come back to this!

**Comment by mec** [comment]
Superseded by #160

**Pull request closed without merging**